### PR TITLE
gh-119698: deprecate ``symtable.Class.get_methods``

### PR DESCRIPTION
**Pull request by picnixz**
Deprecation notice for ``symtable.Class.get_methods`` for 3.14+ (to be merged after #121802 since it is included in this PR).

* Issue: gh-119698

----
📚 Documentation preview 📚: https://cpython-previews--121902.org.readthedocs.build/

### CODE REVIEW

File: Lib/symtable.py
Comment (ZeroIntensity) [inline Lib/symtable.py:240]: Why not just put the import at the top of the file?

Comment (picnixz) [reply Lib/symtable.py:240]: To reduce import time.

Comment (ZeroIntensity) [reply Lib/symtable.py:240]: Right, but that import time is now spent on the method, and is no longer a one-time cost. I'll leave this for someone else to nitpick.

**Review by ZeroIntensity — APPROVED** [review]
LGTM, but I'm not too sure about the `import warnings` in the method body instead of at the module level. I'll let a core dev decide on that.

File: Lib/symtable.py
Comment (picnixz) [reply Lib/symtable.py:240]: The point is to remove this method and the rest of the module should not be penalized because of this. If people want to use something that is *not* deprecated, they may not want the `import warnings`. In addition, this is how it is being done in general in the library (see `typing`, `threading`, and others).

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by picnixz** [comment]
Friendly ping @JelleZijlstra in case you forgot about this one (or maybe if you were not aware of it).

**Review by JelleZijlstra — APPROVED** [review]
Thanks, hadn't seen this PR.

**Pull request merged**
